### ISSUE TITLE
toplevel makefile should check for submodule checkout

### ISSUE THREAD

**Issue by ab9rf** [DFHack/dfhack#2480]
it would be beneficial, especially for first-time builders, if the makefile checked for the existence of the git submodules and report a meaningful error if they're absent. "The source directory does not contain a CMakeLists.txt file" provides little guidance on how to fix this mistake, which many new people will likely make

**Comment by ab9rf** [comment]
hm, looks like there is such a test, but there's so many messages _after_ that that it gets lost in the haze. maybe abort the build at this point?

**Comment by lethosor** [comment]
Yeah, I chose SEND_ERROR instead of FATAL_ERROR for those messages so that you get all the relevant errors at once, instead of having to play whack-a-mole. It does get rather noisy if there are a lot, though. I don't know of a way to set a limit.
